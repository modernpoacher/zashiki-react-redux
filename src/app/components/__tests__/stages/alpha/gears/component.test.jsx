import React from 'react'
import PropTypes from 'prop-types'
import renderer from 'react-test-renderer'

import Gears from '#app/components/stages/alpha/gears/component'

jest.mock('shinkansen-engine/components/signals', () => ({
  OMEGA_PATTERN: 'MOCK OMEGA PATTERN'
}))

function MockLink ({ to, children }) {
  return (
    <a href={to} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape()
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router-dom', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#app/components/stages/alpha/gears/component', () => {
  describe('Always', () => {
    it('renders', () => {
      const component = (
        <Gears />
      )

      expect(renderer.create(component).toJSON())
        .toMatchSnapshot()
    })
  })

  it('renders `reverse`', () => {
    const component = (
      <Gears reverse={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }} />
    )

    expect(renderer.create(component).toJSON())
      .toMatchSnapshot()
  })

  it('renders `forward`', () => {
    const component = (
      <Gears forward={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }} />
    )

    expect(renderer.create(component).toJSON())
      .toMatchSnapshot()
  })

  it('renders `reverse` and `forward`', () => {
    const component = (
      <Gears
        reverse={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
        forward={{ alpha: 'MOCK ALPHA', omega: 'MOCK OMEGA' }}
      />
    )

    expect(renderer.create(component).toJSON())
      .toMatchSnapshot()
  })
})
