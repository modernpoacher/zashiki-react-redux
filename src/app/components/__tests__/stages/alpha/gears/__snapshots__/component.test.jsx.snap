// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#app/components/stages/alpha/gears/component Always renders 1`] = `null`;

exports[`#app/components/stages/alpha/gears/component renders \`forward\` 1`] = `
<ul
  className="shinkansen-gears"
>
  <li
    className="shinkansen-gears-forward"
  >
    <a
      className="mock-link"
      href={
        {
          "pathname": "/MOCK ALPHA/MOCK OMEGA",
        }
      }
    >
      Forward
    </a>
  </li>
</ul>
`;

exports[`#app/components/stages/alpha/gears/component renders \`reverse\` 1`] = `
<ul
  className="shinkansen-gears"
>
  <li
    className="shinkansen-gears-reverse"
  >
    <a
      className="mock-link"
      href={
        {
          "pathname": "/MOCK ALPHA/MOCK OMEGA",
        }
      }
    >
      Reverse
    </a>
  </li>
</ul>
`;

exports[`#app/components/stages/alpha/gears/component renders \`reverse\` and \`forward\` 1`] = `
<ul
  className="shinkansen-gears"
>
  <li
    className="shinkansen-gears-reverse"
  >
    <a
      className="mock-link"
      href={
        {
          "pathname": "/MOCK ALPHA/MOCK OMEGA",
        }
      }
    >
      Reverse
    </a>
  </li>
  <li
    className="shinkansen-gears-forward"
  >
    <a
      className="mock-link"
      href={
        {
          "pathname": "/MOCK ALPHA/MOCK OMEGA",
        }
      }
    >
      Forward
    </a>
  </li>
</ul>
`;
